src/Symfony/Component/AssetMapper/Factory/CachedMappedAssetFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Symfony\Component\AssetMapper\Factory;
+
+use Symfony\Component\AssetMapper\MappedAsset;
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Config\Resource\FileResource;
+
+class CachedMappedAssetFactory implements MappedAssetFactoryInterface
+{
+    public function __construct(
+        private MappedAssetFactoryInterface $innerFactory,
+        private string $cacheDir,
+    )
+    {
+    }
+
+    public function createMappedAsset(string $logicalPath, string $sourcePath): ?MappedAsset
+    {
+        $cachePath = $this->getCacheFilePath($logicalPath, $sourcePath);
+        $configCache = new ConfigCache($cachePath, true);
```

Comment:
should we always enable debug mode ?

Reply:
It's a good question. In practice, this system will never be used in production, because you'll have dumped your `manifest.json` and `importmap.json` files, so only those will be used. 

But, might as well. I've updated it to use `%kernel.debug%`.